frontend/app/src/screens/user-profile/user-profile.tsx

```diff
@@ -77,7 +87,10 @@ export function UserProfilePage() {
     return <LoadingScreen />;
   }
 
-  const profile = data?.AccountProfile;
+  if (!profile || !auth.isAuthenticated) {
+    navigate("/");
+    return;
+  }
```

Comment:
Additionally, I feel it would be beneficial to move the logic for determining "can I access this page" to the `/pages` folder. This way,  components under /screens can focus on display logic based on the data they receive.